tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ITelemeteredConfigurationSource.cs

```diff
@@ -0,0 +1,86 @@
+﻿// <copyright file="ITelemeteredConfigurationSource.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Configuration.ConfigurationSources.Telemetry;
+
+namespace Datadog.Trace.Configuration.Telemetry;
+
+/// <summary>
+/// A version of <see cref="IConfigurationSource"/> that also allows reports the source of the telemetry
+/// when a value is retrieved using a key.
+/// </summary>
+internal interface ITelemeteredConfigurationSource
```

Comment:
How about `ITrackedConfigurationSource` or `IMonitoredConfigurationSource`

Or `IHasTelemetryConfigurationSource`? 😃 

After learning that "telemetered` is a real word, I don't think it is _that_ bad of a name